### DISCUSSION TITLE
Discord Provider not returning all user info/ignoring profile override (Auth.js SvelteKit)

### DISCUSSION THREAD

**Discussion by kamiifx** [nextauthjs/next-auth#7451] [Help]
**Issue** 

_@auth/core: 0.7.1,
@auth/sveltekit: 0.3.1,_

The Discord provider does not seem to return anything more than its initial **[profile](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/discord.ts)**.  Which is _name_, _email_ and _image_ .

**My setup**

I am trying to return additional info that api/user/@me provides like _discriminator_ and _verified_ as seen below.

```js

		providers: [
			Discord({
				clientId: DISCORD_ID,
				clientSecret: DISCORD_SECRET,
				authorization: 'https://discord.com/api/oauth2/authorize?scope=identify+email',
				token: 'https://discord.com/api/oauth2/token',
				userinfo: 'https://discord.com/api/users/@me',
				profile(profile) {
					return {
						id: profile.id,
						name: profile.username,
						email: profile.email,
						image: profile.image_url,
						discriminator: profile.discriminator,
                                                verified: profile.verified
					};
				}
			})
		],

```

Iknow I could do a callback on session and fetch this.

```js
callbacks: {
	async session({ session, user, token }) {
	const userRes = await fetch('https://discord.com/api/users/@me', {
	        headers: {
		authorization: `Bearer ${token.accessToken}`
		}}).then((r) => r.json());
		return { session, userRes };
	}
}
```

But I see the info is already on the profile and available : **[profile](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/discord.ts)** . I might be doing something wrong or missed something, has anyone found a solution or experience the same thing ?

**Full code hooks.server.ts** 

```js
export const handle = SvelteKitAuth(async (event) => {
	return {
		providers: [
			Discord({
				clientId: DISCORD_ID,
				clientSecret: DISCORD_SECRET,
				authorization: 'https://discord.com/api/oauth2/authorize?scope=identify+email',
				token: 'https://discord.com/api/oauth2/token',
				userinfo: 'https://discord.com/api/users/@me',
				profile(profile) {
					return {
						id: profile.id,
						name: profile.username,
						email: profile.email,
						image: profile.image_url,
						discriminator: profile.discriminator
					};
				}
			})
		],
		secret: AUTH_SECRET,
	};
}) satisfies Handle;
```

**Comment by Greek** [comment]
Bumping this. How would we be able to fetch the entire Discord profile *without* fetching `/api/users/@me` on Discord?

**Reply by kamiifx** [reply]
Cause its obv hitting `/api/users/@me` since it returns name and image besides email, there seems to be a deeper issue where you cant override what it returns or something.
